### DISCUSSION TITLE
Using PhotosPickerUI with TCA

### DISCUSSION THREAD

**Discussion by evankline** [pointfreeco/swift-composable-architecture#2729] [Q&A] [answered]
I'm developing an app that allows a user to set their profile image from their photo library and would like to use the [PhotosPicker](https://developer.apple.com/documentation/photokit/photospicker) to achieve this functionality. I'm new to Swift and TCA, so I would like to do this without UIKit.

I'm struggling to understand how to pass the user's selection to the feature's reducer. I've tried a few different approaches but have been unable to figure it out. I can't find an example of this and was hoping someone could point me in the right direction.

**Comment by saroar** [comment, marked as answer]
@evankline check this https://gist.github.com/saroar/819eb7480b7f75010a80e872ee1dee5b may be it will help u

**Reply by evankline** [reply]
Thanks @saroar!
